dashboard/modules/serve/serve_agent.py

```diff
@@ -78,13 +78,57 @@ async def delete_serve_application(self, req: Request) -> Response:
         return Response()
 
     @routes.put("/api/serve/deployments/")
-    @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=True)
+    @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=False)
```

Comment:
hmmm this means if people call /status or GET deployments/, it dashboard will connect to Serve once with the default. And these options here won't be applied because serve already been started.

Reply:
Yeah, this is a potential issue. I added a comment mentioning that users should `DELETE` their existing application if they want to use a new host and port, but I'm not sure if that's sufficient.

On the bright side, this issue should only affect users that want a specific host and port. It shouldn't affect users that don't need this feature.